mlir/lib/Dialect/MIOpen/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -2636,18 +2636,29 @@ struct GridwiseGemmV2RewritePattern
                                       enableOutSwizzles);
     }
 
-    op.erase();
+    b.eraseOp(op);
 
     return success();
   }
 };
 
 void LowerMIOpenOpsStep2Pass::runOnOperation() {
   MLIRContext *ctx = &getContext();
+  ConversionTarget target(*ctx);
+  target.addIllegalOp<miopen::GridwiseGemmOp>();
+  target.addIllegalOp<miopen::GridwiseGemmV2Op>();
+
+  target.addLegalDialect<arith::ArithmeticDialect>();
+  target.addLegalDialect<miopen::MIOpenDialect>();
+  target.addLegalDialect<AffineDialect>();
+
   RewritePatternSet patterns(ctx);
-  patterns.add<GridwiseGemmRewritePattern, GridwiseGemmV2RewritePattern>(ctx);
-  if (failed(applyPatternsAndFoldGreedily(getOperation(), std::move(patterns))))
+  patterns.add<GridwiseGemmRewritePattern>(ctx);
+  patterns.add<GridwiseGemmV2RewritePattern>(ctx);
+  if (failed(applyPartialConversion(getOperation(), target,
```

Comment:
It seems we are only changing the first 2 steps. Do we need to change other lowering steps too?

Reply:
Some of the lower stages actually do rely on the repeated-application/convergency property, like `step4` with `TransformingFor` and the current step3 with `TransformOpRewritePattern`.

Reply:
Correct, we need to change all lowering steps. But if the later stages are dead for any reason, we are doomed anyways (Because we are not allowed to, after applicability checks). So I'd prioritize fixing the first two passes that are actually used in applicability checks.

Reply:
I don't think we need to change the change the rest of the passes

Reply:
> Some of the lower stages actually do rely on the repeated-application/convergency property

Good to know if that's the case. I had no idea that they do so repeatedly.

Reply:
Are you open to fixing up the calls to add() and the like here or do you want to do those later?

Reply:
Done in [72323fd](https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/pull/545/commits/72323fdad1b523f61d1396f8fc875badab6c9aa1)

Reply:
Done in 72323fd